extensions/http-curl/processors/InvokeHTTP.cpp

```diff
@@ -246,6 +246,20 @@ void InvokeHTTP::onSchedule(const std::shared_ptr<core::ProcessContext> &context
   if (context->getProperty(DisablePeerVerification.getName(), disablePeerVerification)) {
     utils::StringUtils::StringToBool(disablePeerVerification, disable_peer_verification_);
   }
+
+  std::string proxy_value;
+  if (context->getProperty(ProxyHost.getName(), proxy_value) && !proxy_value.empty()) {
+    proxy_.host = proxy_value;
+  }
+  if (context->getProperty(ProxyPort.getName(), proxy_value) && !proxy_value.empty()) {
+    proxy_.port = std::stoi(proxy_value);
```

Comment:
`getProperty` is a templated method, that depending on the second argument's type carries out checks and conversions from `std::string`, so feel free to pass `proxy.port` as the second argument here, moreover I believe `getProperty` adheres to the contract "the out argument is not modified on failure, missing value", so we don't even need the "if"s here

Reply:
The other reason to remove if statements is updating the config: we should null the members, otherwise proxy properties can't be removed without restarting minifi.

Reply:
Done in [2bc2b75](https://github.com/apache/nifi-minifi-cpp/pull/897/commits/429675f5d9a526c529080f4468d170e320f21dca)